openedx/core/djangoapps/session_inactivity_timeout/tests/test_middleware.py

```diff
@@ -0,0 +1,209 @@
+"""
+Unit tests for SessionInactivityTimeout middleware.
+"""
+
+from datetime import datetime, timedelta
+from unittest.mock import patch
+
+import ddt
+from django.contrib.auth.models import AnonymousUser
+from django.contrib.sessions.backends.db import SessionStore
+from django.test import TestCase, override_settings
+
+from common.djangoapps.student.tests.factories import UserFactory
+from openedx.core.djangolib.testing.utils import get_mock_request
+from unittest.mock import call
+
+from openedx.core.djangoapps.session_inactivity_timeout.middleware import (
+    SessionInactivityTimeout,
+    LAST_TOUCH_KEYNAME,
+)
+
+
+@ddt.ddt
+class SessionInactivityTimeoutTestCase(TestCase):
+    """
+    Test case for SessionInactivityTimeout middleware
+    """
+    def setUp(self):
+        super().setUp()
+        self.user = UserFactory.create()
+        self.middleware = SessionInactivityTimeout(get_response=lambda request: None)
+        self.request = get_mock_request(self.user)
+
+        self.request.session = SessionStore()
+        self.request.session.create()
+        self.request.session.modified = False
+
+    def test_process_request_unauthenticated_user_does_nothing(self):
+        self.request.user = AnonymousUser()
+        response = self.middleware.process_request(self.request)  # lint-amnesty, pylint: disable=assignment-from-none
+        assert response is None
+        assert LAST_TOUCH_KEYNAME not in self.request.session
+
+    @ddt.data(None, 0)
+    def test_process_request_timeout_disabled_does_nothing(self, timeout_value):
+        with override_settings(SESSION_INACTIVITY_TIMEOUT_IN_SECONDS=timeout_value):
+            response = self.middleware.process_request(self.request)  # lint-amnesty, pylint: disable=assignment-from-none
+
+        assert response is None
+        assert LAST_TOUCH_KEYNAME not in self.request.session
+
+    @ddt.data(
+        None,  # No timestamp key in session
+        "",    # Empty string timestamp
```

Comment:
Consistency here will make this more clear.
```suggestion
        "",    # Empty string timestamp in session
```

Reply:
updated